sshauthentication-api/src/main/java/org/openintents/ssh/authentication/util/key/SshEd25519PublicKey.java

```diff
@@ -64,11 +66,10 @@ public void setAByte(byte[] aByte) {
     // TODO: remove?
     // encode point according to RFC8032 section 5.1.5
     private byte[] encodePoint() {
-        byte[] yIn = mY.toByteArray();
-        byte[] y = new byte[32];
-        for (int i = 0; i < 32; i++) {
-            y[i] = yIn[31 - i]; // convert to little endian
-        }
+        ByteBuffer buffer = ByteBuffer.allocate(32).order(ByteOrder.LITTLE_ENDIAN);
```

Comment:
`ByteBuffer.wrap(mY.toByteArray()).order(ByteOrder.LITTLE_ENDIAN).array()` not tested but I'd expect that to work

Reply:
Neither work. After some trial&error I noticed that ByteBuffer doesn't have a concept of little-endianness when it comes to byte[] (since that is a bulk operation). This really requires a loop since little-endian BigIntegers are not a thing in Java apparently.

Reply:
I have a test for this function with point data extracted from a public key blob produced by OpenSSH and I'm fairly certain it is correct :). I have not included it since it adds junit&robolectric as dependecies to the api library.